main/pipeline/pipeline.cc

```diff
@@ -869,6 +870,8 @@ vector<ast::ParsedFile> resolve(unique_ptr<core::GlobalState> &gs, vector<ast::P
                 ENFORCE(reresolved.size() == 1);
                 f = checkNoDefinitionsInsideProhibitedLines(*gs, move(reresolved[0]), 0, prohibitedLines);
             }
+            ENFORCE(symbolsBefore == gs->symbolsUsed(),
```

Comment:
This could be a more intensive test that checks for full equality of the symbol table, and not just whether it's the same size, but this is a cheap test that should still be true.

Reply:
nice